File: langwatch/src/features/onboarding/regions/observability/codegen/snippets/python/anthropic.snippet.py
import langwatch  # +
from anthropic import Anthropic
import os

from openinference.instrumentation.anthropic import AnthropicInstrumentor  # +

langwatch.setup(instrumentors=[AnthropicInstrumentor()])  # +

client = Anthropic(api_key=os.getenv("ANTHROPIC_API_KEY"))


@langwatch.trace(name="Anthropic Call with Community Instrumentor")
def generate_text_with_community_instrumentor(prompt: str):
    response = client.messages.create(
        model="claude-sonnet-4-5-20250929",
        max_tokens=1024,
        messages=[{"role": "user", "content": prompt}],
    )
    return response.content[0].text


if __name__ == "__main__":
    user_query = "Tell me a joke"
    response = generate_text_with_community_instrumentor(user_query)
    print(f"User: {user_query}")
    print(f"AI: {response}")

File: langwatch/src/features/onboarding/regions/observability/codegen/snippets/go/anthropic.snippet.go
package main

import (
	"context"
	"log"
	"os"

	"github.com/anthropics/anthropic-sdk-go"
	anthropicoption "github.com/anthropics/anthropic-sdk-go/option"
	langwatch "github.com/langwatch/langwatch/sdk-go"                               // +
	otelanthropic "github.com/langwatch/langwatch/sdk-go/instrumentation/anthropic" // +
	"go.opentelemetry.io/otel"                                                      // +
	sdktrace "go.opentelemetry.io/otel/sdk/trace"                                   // +
)

func main() {
	ctx := context.Background()

	// Setup LangWatch tracing (reads LANGWATCH_API_KEY from env) // +
	exporter, err := langwatch.NewExporter(ctx) // +
	if err != nil {                             // +
		log.Fatalf("failed to create LangWatch exporter: %v", err) // +
	} // +
	tp := sdktrace.NewTracerProvider(sdktrace.WithBatcher(exporter)) // +
	otel.SetTracerProvider(tp)                                       // +
	defer tp.Shutdown(ctx)                                           // +

	client := anthropic.NewClient(
		anthropicoption.WithAPIKey(os.Getenv("ANTHROPIC_API_KEY")),
		anthropicoption.WithMiddleware(otelanthropic.Middleware()), // +
	)

	message, err := client.Messages.New(ctx, anthropic.MessageNewParams{
		Model:     anthropic.ModelClaudeSonnet4_5,
		MaxTokens: 1024,
		System:    []anthropic.TextBlockParam{{Text: "You are a helpful assistant."}},
		Messages: []anthropic.MessageParam{
			anthropic.NewUserMessage(anthropic.NewTextBlock("Hello, Claude!")),
		},
	})
	if err != nil {
		log.Fatalf("Messages call failed: %v", err)
	}

	log.Printf("Response: %v", message)
}
